src/main/java/net/glowstone/io/anvil/RegionFile.java

```diff
@@ -200,19 +214,19 @@ public DataInputStream getChunkDataInputStream(int x, int z) throws IOException
         if (version == VERSION_GZIP) {
             byte[] data = new byte[length - 1];
             file.read(data);
-            return new DataInputStream(new GZIPInputStream(new ByteArrayInputStream(data)));
+            return new DataInputStream(new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(data))));
```

Comment:
I don't think I was being clear enough. Let me rephrase what I said earlier.

`InflaterInputStream` includes a built-in buffer so it can read data from the backing input stream, which in our case is a `ByteArrayInputStream`. `InflaterInputStream.read()` is implemented by inflating this buffer into an user-provided byte array.

Personally, I think it's better CPU-wise to decompress a few large chunks of data at a time than decompress a lot of small chunks.